DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs

```diff
@@ -0,0 +1,217 @@
+    {
+        public IEnumerable<CurrentSelfAssessment> GetSelfAssessmentsForCandidate(int candidateId)
+        {
+            return connection.Query<CurrentSelfAssessment>(
+                @"SELECT
+                        CA.SelfAssessmentID AS Id,
+                        SA.Name,
+                        SA.Description,
+                        SA.IncludesSignposting,
+                        SA.SupervisorResultsReview AS IsSupervisorResultsReviewed,
+                        COALESCE(SA.Vocabulary, 'Capability') AS Vocabulary,
+                        COUNT(C.ID) AS NumberOfCompetencies,
+                        CA.StartedDate,
+                        CA.LastAccessed,
+                        CA.CompleteByDate,
+                        CA.UserBookmark,
+                        CA.UnprocessedUpdates,
+                        CA.LaunchCount,
+                        1 AS IsSelfAssessment,
+                        CA.SubmittedDate
+                    FROM CandidateAssessments CA
+                    JOIN SelfAssessments SA
+                        ON CA.SelfAssessmentID = SA.ID
+                    INNER JOIN SelfAssessmentStructure AS SAS
+                        ON CA.SelfAssessmentID = SAS.SelfAssessmentID
+                    INNER JOIN Competencies AS C
+                        ON SAS.CompetencyID = C.ID
+                    WHERE CA.CandidateID = @candidateId AND CA.RemovedDate IS NULL AND CA.CompletedDate IS NULL
+                    GROUP BY
+                        CA.SelfAssessmentID, SA.Name, SA.Description, SA.IncludesSignposting, SA.SupervisorResultsReview,
+                        COALESCE(SA.Vocabulary, 'Capability'), CA.StartedDate, CA.LastAccessed, CA.CompleteByDate,
+                        CA.UserBookmark, CA.UnprocessedUpdates, CA.LaunchCount, CA.SubmittedDate",
+                new { candidateId }
+            );
+        }
+
+        public CurrentSelfAssessment? GetSelfAssessmentForCandidateById(int candidateId, int selfAssessmentId)
+        {
+            return connection.QueryFirstOrDefault<CurrentSelfAssessment>(
+                @"SELECT
+                        CA.SelfAssessmentID AS Id,
+                        SA.Name,
+                        SA.Description,
+                        SA.QuestionLabel,
+                        SA.DescriptionLabel,
+                        SA.IncludesSignposting,
+                        SA.SupervisorResultsReview AS IsSupervisorResultsReviewed,
+                        COALESCE(SA.Vocabulary, 'Capability') AS Vocabulary,
+                        COUNT(C.ID) AS NumberOfCompetencies,
+                        CA.StartedDate,
+                        CA.LastAccessed,
+                        CA.CompleteByDate,
+                        CA.UserBookmark,
+                        CA.UnprocessedUpdates,
+                        CA.LaunchCount, CA.SubmittedDate, SA.LinearNavigation, SA.UseDescriptionExpanders, SA.ManageOptionalCompetenciesPrompt,

```

Comment:
Is there a reason these are all on one line?

Reply:
Nope, just missed them in the messy SQL